### PR TITLE
[rebased]: add method get_databases to meta_api

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

This is the same as PR #2019 (except rebased)

`meta_api`:

- new method `get_databases`
- new method `get_tables`
- rm method `get_database_meta`

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

None

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
I think this patch is ready for review :)

**Comment by dantengsky** [comment]
> I think this patch is ready for review :)

yes, forgot to convert it to "ready for review", thanks for reminding me

File: kvsrv/src/meta_service/raftmeta.rs [hunk 1]
```diff
@@ -1008,32 +1008,68 @@ impl MetaNode {
         sm.get_database(name)
     }
 
+    //    #[tracing::instrument(level = "debug", skip(self))]
```
Comment (BohuTANG) [inline kvsrv/src/meta_service/raftmeta.rs:1011]: This codes no used anymore?

Comment (dantengsky) [reply kvsrv/src/meta_service/raftmeta.rs:1011]: ooops, forgot to clean them

Comment (dantengsky) [reply kvsrv/src/meta_service/raftmeta.rs]: cleaned, PTAL

**Comment by BohuTANG** [comment]
/lgtm

Sorry for the conflicts merge

File: query/src/catalogs/mod.rs [hunk 2]
```diff
@@ -17,11 +17,9 @@ pub use catalog::Catalog;
 pub use database::Database;
 pub use impls::util::in_memory_metas::InMemoryMetas;
 pub use meta_id_ranges::*;
-//pub use meta_backend::MetaBackend;
```
Comment (BohuTANG) [inline query/src/catalogs/mod.rs:20]: I guess this can be removed in the next patch

**Comment by dantengsky** [comment]
> /lgtm
> 
> Sorry for the conflicts merge

don't worry

File: query/src/catalogs/mod.rs (hunk 2 above)
Comment (dantengsky) [reply query/src/catalogs/mod.rs:20]: OK, will be clean in PR #2026

**Review by drmingdrmer — APPROVED** [review]

**Pull request merged**